wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/WireRun.kt

```diff
@@ -170,9 +171,40 @@ data class WireRun(
    */
   val targets: List<Target>,
 
+  /**
+   * A map from module dir to module info which dictates how the loaded types are partitioned and
+   * generated.
+   *
+   * By default a single module is used which includes everything in the root output directory.
+   * If desired, multiple modules can be specified along with dependencies between them. Types
+   * which appear in dependencies will not be re-generated.
+   */
+  val modules: Map<String, Module> = mapOf("./" to Module()),
+
   /** True to build proto3 artifacts. This is unsupported and does not work. */
   val proto3Preview: Boolean = false
 ) {
+  data class Module(
+    val dependencies: Set<String> = emptySet(),
+    val pruningRules: PruningRules? = null
+  )
+
+  init {
+    val dagChecker = DagChecker(modules.keys) { moduleName ->
```

Comment:
For follow up, we can move `check()` into the new `DirectedAcyclicGraph` class!

Reply:
Yeah it lives in wire-schema and this in wire-compiler. I can move this up into wire-schema and combine afterward.